Automattic/wp-calypso
# Record build target served.

We currently don't keep track of how many requests get served with the fallback vs the evergreen target. This change adds a count for each.

To achieve this, it also adds the ability to record counting-type stats.

**Question**: is recording `target-fallback` and `target-evergreen` enough, or do we also need a total in order to determine % of fallback and % of evergreen?

#### Changes proposed in this Pull Request

* Convert `lib/analytics/statsd` to TypeScript, refactor, and add `statsdCountingUrl` method
* Extract feature slug composition into its own module
* Add `recordCounting` method to both server-side and client-side analytics libs
* Add build target to context
* Add build target counting to server-side logging
* Add unit tests for new stuff

#### Testing instructions

* Ensure that response time logging is unaffected
* Ensure that `target-evergreen` gets incremented for evergreen requests (use e.g. the latest Chrome or Firefox)
* Ensure that `target-fallback` gets incremented for fallback requests (use an old browser or append `?forceFallback=1` to the URL
